packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton/index.tsx

```diff
@@ -43,6 +68,31 @@ export const RainbowKitCustomConnectButton = () => {
 
               return (
                 <>
+                  <div className="flex items-center gap-2">
+                    {isLoading ? (
+                      <span>Checking membership...</span>
+                    ) : isMember ? (
+                      <span title="Batch member" className="text-green-500">
+                        ✅ Member
+                      </span>
+                    ) : (
+                      <span title="Not a batch member" className="text-red-500">
+                        ❌ Not a member
+                      </span>
+                    )}
+
+                    {isLoading ? (
+                      <span>Checking check-in...</span>
+                    ) : checkedInContract && checkedInContract !== "0x0000000000000000000000000000000000000000" ? (
+                      <span title="Already checked in" className="text-blue-500">
+                        🟢 Checked in
+                      </span>
+                    ) : (
+                      <span title="Not checked in" className="text-gray-400">
+                        ⚪ Not checked in
+                      </span>
+                    )}
+                  </div>
```

Comment:
To save space, I think we could have just one `status` instead of two, with the following states:

* Loading
* Not a member
* Not checked in
* Checked in

Oh, and remember to handle the case when the user is not logged in.